### PR TITLE
TST: parametrize arithmetic tests

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/tests/arithmetic/test_datetime64.py [hunk 1]
```diff
@@ -30,6 +30,45 @@
 import pandas.util.testing as tm
 
 
+def assert_invalid_comparison(left, right, box):
+    # Not for tznaive-tzaware comparison
```
Comment (TomAugspurger) [inline pandas/tests/arithmetic/test_datetime64.py]: Should this be checked, raising if it's not the case?

File: pandas/tests/arithmetic/test_period.py [hunk 2]
```diff
@@ -573,12 +573,19 @@ def test_parr_add_sub_float_raises(self, op, other, box_with_array):
...
+            # datetime scalars
             pd.Timestamp.now(),
             pd.Timestamp.now().to_pydatetime(),
             pd.Timestamp.now().to_datetime64(),
+            # datetime-like arrays
+            pd.date_range("2016-01-01", periods=3, freq="H"),
+            pd.date_range("2016-01-01", periods=3, tz="Europe/Brussels"),
+            pd.date_range("2016-01-01", periods=3, freq="S")._data,
+            pd.date_range("2016-01-01", periods=3, tz="Asia/Tokyo")._data,
+            # Miscellaneous invalid types
```
Comment (TomAugspurger) [inline pandas/tests/arithmetic/test_period.py]: Is this comment still relevant?

File: pandas/tests/arithmetic/test_datetime64.py [hunk 3]
```diff
@@ -30,6 +30,45 @@
 import pandas.util.testing as tm
 
 
+def assert_invalid_comparison(left, right, box):
```
Comment (TomAugspurger) [inline pandas/tests/arithmetic/test_datetime64.py]: Is this datetime-specific? Or should (/could) it be moved to pandas.util.testing?

**Review by TomAugspurger — COMMENTED** [review]
LGTM overall. Small questions that can be ignored.

File: pandas/tests/arithmetic/test_datetime64.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/arithmetic/test_datetime64.py]: I don't think so.  If this is used incorrectly, this call will fail

File: pandas/tests/arithmetic/test_period.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/arithmetic/test_period.py]: Yes.  Pre-black formatting there was a newline before this comment to clarify the separation

File: pandas/tests/arithmetic/test_datetime64.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/tests/arithmetic/test_datetime64.py]: Potentially (there are a few assertion helper methods in this directory that might).  The main drawback is that if you pass a Series or DataFrame it only works with default name/index/columns

Comment (jreback) [reply pandas/tests/arithmetic/test_datetime64.py]: can you add a doc-string / types

**Review by jreback — COMMENTED** [review]
lgtm.

**Comment by jreback** [comment]
minor comments, ping on green

**Comment by jbrockmendel** [comment]
ping

**Comment by jbrockmendel** [comment]
@jreback if this and #27807 can get through before you head out of town I can keep busy with further test parametrization/de-duplication

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**